### ISSUE TITLE
Create consistent, logical anchors in User Guide

### ISSUE THREAD

**Issue by sbrannen** [junit-team/junit5#320]
## Overview

Since it is highly unprofessional to allow anchors to be auto-generated by AsciiDoctor based on section headers (i.e., because somebody might randomly change the wording thereby breaking bookmarked links), we must create consistent, logical anchors for every section in the User Guide.

## Example

Some work in this direction has already been started in the _Extension Model_ -- for example:

```
[[extension-model]]
== Extension Model

[[extension-registration]]
=== Registering Extensions

[[extension-registration-declarative]]
==== Declarative Extension Registration
```

**Comment by hglasgow** [comment]
I've created PR #355, adding unique section header anchors to the documentation where they were missing. I have not altered any that were already there.

**Comment by sbrannen** [comment]
PR #355 has been merged into `master` in 8f87e4a0cf1ca3216cec3427bf8753b8d4d6aa99.

Cheers!
